algorithms/src/polycommit/sonic_pc/data_structures.rs

```diff
@@ -476,15 +575,52 @@ impl<E: PairingEngine> VerifierKey<E> {
             .as_ref()
             .and_then(|v| v.binary_search_by(|(d, _)| d.cmp(&degree_bound)).ok().map(|i| v[i].1.clone()))
     }
-}
 
-impl<E: PairingEngine> VerifierKey<E> {
     pub fn max_degree(&self) -> usize {
         self.max_degree
     }
 
-    pub fn supported_degree(&self) -> usize {
-        self.supported_degree
+    pub fn union<T: IntoIterator<Item = &'a VerifierKey<E>>>(verifier_keys: T) -> Self {
+        let mut bounds_seen = HashSet::<usize>::new();
+        let mut bounds_and_neg_powers = vec![];
+        let mut bounds_and_prepared_neg_powers = vec![];
+        let mut biggest_vk: Option<&VerifierKey<E>> = None;
+        for vk in verifier_keys {
+            if biggest_vk.is_none() || biggest_vk.unwrap().supported_degree < vk.supported_degree {
+                biggest_vk = Some(vk);
+            }
+            let new_bounds = vk.degree_bounds_and_neg_powers_of_h.as_ref().unwrap();
+            let new_prep_bounds = vk.degree_bounds_and_prepared_neg_powers_of_h.as_ref().unwrap();
+            assert_eq!(new_bounds.len(), new_prep_bounds.len());
+            for ((bound, neg_powers), (bound2, prep_neg_powers)) in new_bounds.iter().zip(new_prep_bounds) {
+                assert_eq!(bound, bound2);
+                if bounds_seen.insert(*bound) {
+                    bounds_and_neg_powers.push((*bound, neg_powers));
+                    bounds_and_prepared_neg_powers.push((*bound, prep_neg_powers));
+                }
+            }
+        }
+
+        let biggest_vk = biggest_vk.unwrap();
+        let mut union = VerifierUnionKey::<E> {
```

Comment:
`union` is a [reserved keyword](https://doc.rust-lang.org/std/keyword.union.html)

Reply:
7d1765266